stellar_resources/src/utils.rs

```diff
@@ -0,0 +1,26 @@
+use serde::{Deserialize, Deserializer};
+use serde::de;
+use std::str::FromStr;
+
+pub fn deserialize_from_str<'de, D, T>(d: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
```

Comment:
Why does this require a lifetime?

Reply:
I think so that the return value has the same lifetime as the caller.

Reply:
Interesting.  It catches me a bit off guard as it's being passed in as an argument and so I would expect it to have that scope out of the box but hey the compiler doesn't lie.  Let's ship this beast.

Reply:
I can try it without it, the type signature from the documentation required it.

Reply:
Ok, so the issue is that the `Deserializer` trait takes a life time parameter, there for it's needed. It must contain a reference internally.

Reply:
https://github.com/serde-rs/serde/blob/master/serde/src/de/mod.rs#L497

So the trait requires it so it must be declared in the function signature. I think it's because it wasn't to be able to do "in place" deserialization, which would require keeping the references bound to the same lifetime.